src/main/java/org/junitpioneer/jupiter/IssueExtension.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.platform.commons.support.AnnotationSupport;
+
+/**
+ * This class implements the functionality for the {@code @Issue} annotation.
+ *
+ * @see Issue
+ */
+@DisplayName("Issue extension")
+class IssueExtension implements BeforeEachCallback {
+
+	static final Namespace NAMESPACE = Namespace.create(IssueExtension.class);
+	static final String KEY = "Issue";
+
+	@Override
+	public void beforeEach(ExtensionContext context) {
+		//noinspection unchecked
+		if (PioneerAnnotationUtils.isAnyAnnotationPresent(context, Issue.class)) {
+			storeIssueId(context);
+		}
+	}
+
+	/**
+	 * Reads the {@code @Issue} value from the annotation and publishes it in the extension context.
+	 * @param context The Extensions context
+	 */
+	void storeIssueId(ExtensionContext context) {
```

Comment:
Name was also a leftover - good spot!
Yeah with the new implementation a separate method only for publishing is overkill and doesn't improve readability (like it was before). So I removed the `storeIssueId` method. I want to keep the method to read the value from the annotation, because for me the code is more readable when its in an own method.